### ISSUE TITLE
Disconnect hang 2.6.0.4

### ISSUE THREAD

**Issue by tinohager** [jchristn/SuperSimpleTcp#124]
With the current nuget package the logic hangs on disconnect.
I have attached 2 test clients for this (WinForm, WPF)

[Test.SuperSimpleTcp.zip](https://github.com/jchristn/SuperSimpleTcp/files/8115455/Test.SuperSimpleTcp.zip)

**Comment by tinohager** [comment]
The originator of the error is from my point of view this call

https://github.com/jchristn/SuperSimpleTcp/blob/1e92103d7c5051f9e4308cddd932cd6d67cad360/SuperSimpleTcp/SimpleTcpClient.cs#L445

Also, the `Register()` should be before the while loop which was also wrong in my example.

https://github.com/jchristn/SuperSimpleTcp/blob/1e92103d7c5051f9e4308cddd932cd6d67cad360/SuperSimpleTcp/SimpleTcpClient.cs#L640

**Comment by jchristn** [comment]
Hi @tinohager the reason for the ```.Wait()``` is to ensure the ```DataReceiver``` exits before allowing ```Disconnect``` to finish.  My concern is that (and this was happening to another user that filed a bug on it) the user can't immediately call ```Connect``` after calling ```Disconnect``` because the cancellation had not completed.  

Adding the ```.Register``` change now!

**Comment by jchristn** [comment]
Like this, correct?
![image](https://user-images.githubusercontent.com/1421081/155171130-903469ad-1a75-4755-a621-d74d9f62cedb.png)

**Comment by tinohager** [comment]
I have now tested again with my own implementation. From my point of view the register must be executed at the initialize of the CancellationToken. A one-time registration should be enough as long as the CancellationToken is not triggered.

```cs
token.Register(() =>
{
	if (this._stream == null)
	{
		return;
	}

	this._stream.Close();
});
```

**Comment by jchristn** [comment]
Just published v2.6.0.5 to NuGet: https://www.nuget.org/packages/SuperSimpleTcp/2.6.0.5
Commit: https://github.com/jchristn/SuperSimpleTcp/commit/c54bb6433e701f25e53dbb166f903912071556b0

I tested on .NET 4.6.1, .NET 4.8, .NET Core 3.1, .NET 5.0, and .NET 6.0.  Everything seems to be working as it should.  Please re-open if you encounter this issue again!  Thanks again @tinohager for all of your help!

**Comment by donslattery** [comment]
I'm having the same issue with the hanging on disconnect.  I'm using a test socket utility called Hercules since I don't have direct access to the PLC automation controller we use at our facility all the time.  Once the connection is established, I send some data back and forth which works well.  Then calling the Disconnect() method on SimpleTcpClient waits seemingly forever.  Even killing the connection on the socket side of the test utility makes no difference.
